src/flamenco/runtime/fd_blockstore.c

```diff
@@ -561,17 +559,62 @@ fd_blockstore_buffered_shreds_remove( fd_blockstore_t * blockstore, ulong slot )
         ele = fd_buf_shred_map_ele_remove( shred_map, &key, NULL, shred_pool ) ) )
       fd_buf_shred_pool_ele_release( shred_pool, ele );
   }
-  fd_block_map_remove( slot_map, &slot );
+  fd_block_map_map_ele_remove( block_map, &slot, NULL, slot_pool );
   return FD_BLOCKSTORE_OK;
 }
 
 int
-fd_blockstore_publish( fd_blockstore_t * blockstore, ulong smr ) {
-  long  prune_time_ns    = -fd_log_wallclock();
-  ulong prune_cnt  = 0UL;
+fd_blockstore_block_map_evict( fd_blockstore_t * blockstore, ulong slot ) {
```

Comment:
can be static and private to the blockstore. can just be a `static void prune(...)` to keep naming consistent

no need for `slot` param -- that will always be blockstore->smr